homeassistant/components/apple_tv/remote.py

```diff
@@ -41,6 +52,24 @@ async def async_setup_entry(
     async_add_entities: AddConfigEntryEntitiesCallback,
 ) -> None:
     """Load Apple TV remote based on a config entry."""
+    platform = entity_platform.async_get_current_platform()
+    # Register the services
+    platform.async_register_entity_service(
+        SERVICE_SET_SEARCH_TEXT,
+        {vol.Required(conversation.ATTR_TEXT): cv.string},
+        "async_set_search_text",
+    )
+    platform.async_register_entity_service(
+        SERVICE_APPEND_SEARCH_TEXT,
+        {vol.Required(conversation.ATTR_TEXT): cv.string},
+        "async_append_search_text",
+    )
+    platform.async_register_entity_service(
+        SERVICE_CLEAR_SEARCH_TEXT,
+        None,
+        "async_clear_search_text",
+    )
```

Comment:
Okay so since the atv object is a shared one, we can use a normal action. The big benefit of a normal action of an entity action, is that we can verify the automations that use them. For more info please check https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/action-setup